Core/clim-core/table-formatting.lisp

```diff
@@ -54,6 +54,94 @@
 
 (in-package :clim-internals)
 
+;;; Macro argument checking
+
+(eval-when (:compile-toplevel :load-toplevel :execute)
+  ;; FORMAT-ITEMS uses FORMATTING-CELL, so we need this function at
+  ;; compile-time.
+  (defun maybe-check-constant-keyword-arguments
+      (record-type standard-record-type arguments valid-keywords)
+    (when (and (constantp record-type)
+               (eq (eval record-type) standard-record-type))
+      (loop for (key value) on arguments by #'cddr
+            when (constantp key)
+            do (let ((key (eval key)))
+                 (unless (or (eq key :record-type)
+                             (member key valid-keywords))
+                   (with-current-source-form (arguments)
+                     (error "~@<~S is not a valid initarg for ~S.~@:>"
+                            key standard-record-type))))))))
+
+;;; Helper macro for defining FORMATTING-* macros in the corresponding
+;;; INVOKE-FORMATTING-* functions.
+
```

Comment:
explain in the comment binding optinons (bind, default, transform and passp)

Reply:
Done.